src/app/panels/middle/xvb.rs

```diff
@@ -23,235 +30,362 @@ use crate::{
+                // also change hero mode of runtime.
+                lock!(api).stats_priv.runtime_mode = RuntimeMode::Hero;
+            } else {
+                lock!(api).stats_priv.runtime_mode = RuntimeMode::Auto;
+            }
+
+        }
+    });
+
+
         ui.add_space(space_h);
-        ui.horizontal(|ui| {
-        // hovering text is difficult because egui doesn't hover over inner widget. But on disabled does.
-                ui.group(|ui| {
-                    ui.colored_label(color, text)
-                    .on_hover_text(XVB_HELP);
-                    ui.add(
-                        TextEdit::singleline(&mut self.token)
-                            .char_limit(9)
-                            .desired_width(ui.text_style_height(&TextStyle::Body) * 9.0)
-                            .vertical_align(egui::Align::Center),
-                        ).on_hover_text(XVB_HELP)
+
+         // --------------------------- XVB Advanced -----------------------------------------
+         if !self.simple {
+
+            ui.group(|ui| {
+                ui.vertical_centered(|ui| {
+                    ui.horizontal(|ui| {
+
+                        egui::ComboBox::from_label("")
+                        .selected_text(format!("{:?}", self.mode))
+                        .show_ui(ui, |ui| {
+                                ui.selectable_value(&mut self.mode, XvbMode::Auto, "Auto");
+                                ui.selectable_value(&mut self.mode, XvbMode::Hero, "Hero");
+                                ui.selectable_value(&mut self.mode, XvbMode::ManualXvb, "ManualXvb")
+                                .on_hover_text(XVB_MODE_MANUAL_XVB_HELP);
+                                ui.selectable_value(&mut self.mode, XvbMode::ManualP2pool, "ManualP2pool")
+                                .on_hover_text(XVB_MODE_MANUAL_P2POOL_HELP);
+                                ui.selectable_value(&mut self.mode, XvbMode::ManualDonationLevel, "ManualDonationLevel")
+                                .on_hover_text(XVB_MODE_MANUAL_DONATION_LEVEL_HELP);
+                        });
+                        if self.mode == XvbMode::ManualXvb || self.mode == XvbMode::ManualP2pool {
+
+                            ui.add_space(space_h);
+
+                            let mut is_slider_enabled = true;
+                            let mut hashrate_xmrig = {
+                                if lock!(gui_api_xmrig).hashrate_raw_15m > 0.0 {
+                                    lock!(gui_api_xmrig).hashrate_raw_15m
+                                } else if lock!(gui_api_xmrig).hashrate_raw_1m > 0.0 {
+                                    lock!(gui_api_xmrig).hashrate_raw_1m
+                                } else if lock!(gui_api_xmrig).hashrate_raw > 0.0 {
+                                    lock!(gui_api_xmrig).hashrate_raw
+                                } else {
```

Comment:
slider should still be enabled here.

Reply:
I am not sure about this one. Why should the slider be enabled before xmrig is started. Right now the slider will get enabled as soon as xmrig reports any hashrate above 0. Why should the slider be enabled before this? Before this point in time the slider shows 1000 hashrate which is not the real hashrate. I don't see any point in the slider being enabled before xmrig reports the hashrate.

Reply:
> I don't see any point in the slider being enabled before xmrig reports the hashrate.

We want the user to be able to set it without anything started yet, as he can know in advance how much HR he will get and doesn't want to wait the start of xmrig. They would just have to launch Gupaxx with everything auto starting.

 I thought that's what we talked about [before](https://github.com/Cyrix126/gupaxx/pull/11#issuecomment-2161117694) but I think I didn't state clearly it was not only before xvb but also before xmrig. If the value set is too much when real HR appear, it will be overwritten for runtime but not saved.

> And adding a hovering help message so users understand that their set value can change if it's too much for xmrig.

That's why.


> Touching the slider before the hashrate starts coming in also overrides the configuration from previous sessions.

It should not be saved unless "Save" button is clicked on but should be applied to next algo anyway (same as current hero mode behavior).

Also I just realized that the HR values do not get cleaned when xmrig is stopped. I fixed it in https://github.com/Cyrix126/gupaxx/commit/5c1266cba2bb94fb94c7602ef5f19205400ebf12.

Reply:
Sorry, I had misunderstood your point previously. That makes sense.